mypy/build.py

```diff
@@ -2841,20 +2847,21 @@ def load_graph(sources: List[BuildSource], manager: BuildManager,
         # - In this case A's cached *direct* dependencies are still valid
         #   (since direct dependencies reflect the imports found in the source)
         #   but A's cached *indirect* dependency on C is wrong.
-        dependencies = [dep for dep in st.dependencies if st.priorities.get(dep) != PRI_INDIRECT]
+        dependencies = [dep for dep in st.dependencies if st.priorities.get(dep) != PRI_INDIRECT] \
+            if st.dependencies else []
         if not manager.use_fine_grained_cache():
             # TODO: Ideally we could skip here modules that appeared in st.suppressed
             # because they are not in build with `follow-imports=skip`.
             # This way we could avoid overhead of cloning options in `State.__init__()`
             # below to get the option value. This is quite minor performance loss however.
-            added = [dep for dep in st.suppressed if find_module_simple(dep, manager)]
+            added = [dep for dep in st.suppressed if find_module_simple(dep, manager)] if st.suppressed else []
         else:
             # During initial loading we don't care about newly added modules,
             # they will be taken care of during fine grained update. See also
             # comment about this in `State.__init__()`.
             added = []
-        for dep in st.ancestors + dependencies + st.suppressed:
-            ignored = dep in st.suppressed_set and dep not in entry_points
+        for dep in (st.ancestors or []) + (dependencies or []) + (st.suppressed or []):
+            ignored = dep in st.suppressed_set and dep not in entry_points if st.suppressed_set else []
```

Comment:
- `st.ancestors` can't be None, because there's `assert` above
- `dependencies` can't be None, because it comes from list comprehension or `[]`

Are you sure that all other changes here are needed? This also applies to checks below.

Also, `ignored` is intended to be a boolean, not a list.